scripts/bfasst/compare/waveform.py

```diff
@@ -73,536 +53,121 @@ def compare_netlists(self, design):
 
         self.print_running_compare()
 
-        impl_path = design.impl_netlist_path
-        module = impl_path.name[0 : len(impl_path.name) - 7]
-        build_dir = self.work_dir
-        diff = build_dir / "diff.txt"
-        impl_vcd = build_dir / (module + "_impl.vcd")
-        reversed_vcd = build_dir / (module + "_reversed.vcd")
-
-        shutil.copyfile(impl_path, build_dir / design.impl_netlist_path.name)
-        shutil.copyfile(design.reversed_netlist_path, build_dir / design.reversed_netlist_path.name)
-
-        #print(self.work_dir)
-
-        #Added this in to check if there are multiple verilog files or not. This will present the user with a warning because
-        #currently these can cause strange errors with equivalence checking. 
-        with open(design.yaml_path) as fp:
-            design_props = yaml.safe_load(fp)
-        
-        giveWarning = False
-
-        for k,v in design_props.items():
-            # Handle 'include_all_verilog_files' option
-            if k == "include_all_verilog_files":
-                giveWarning = True
-            # Handle 'include_all_system_verilog_files' option
-            elif k == "include_all_system_verilog_files":
-                giveWarning = True
-
-        if giveWarning:
-            cont = input("Warning: Detected multiple verilog files. Results may be inaccurate.\n Continue anyways? 1 for yes, 0 for no.")
-            if(cont == "0"):
-                print("Ok. Returning non-equivalent status.")
-                return Status(CompareStatus.NOT_EQUIVALENT)
-
-        #This series of if/else statements is used to check if the tests have already been performed. If they have, an option is presented
-        # for the user to view the previously-generated waveforms. If the design was unequivalent previously, the diff output will be
-        # pasted into the linux terminal so they can view specific times in the waveform.
+        paths = get_paths.get_paths(
+            self, design
```

Comment:
self should never be passed explicitly.  If a function requires self, it should be part of the class.  In addition, it looks like paths is dependent on the filepaths defined in self, so that should also be made part of the class.

Reply:
This is the solution I came to from my understanding of the comment: The only aspect of paths that requires self is the actual build directory. I could just pass in the directory itself like get_paths(self.work_dir, design) and then it creates all of the paths from there. If I'm incorrect on this, I may have misunderstood the comment.

Reply:
That approach works.  My initial thought was to bring it back into the class, but I think I like this better.  You could also avoid creating paths separately by just creating it in get_paths, and then returning it.  Anyway, this architecture is much better :)